app/src/main/java/app/magic/wilson/zach/com/magicappkotlin/api/ApiClient.kt

```diff
@@ -1,19 +1,29 @@
 package app.magic.wilson.zach.com.magicappkotlin.api
 
-import android.util.Log
-import app.magic.wilson.zach.com.magicappkotlin.models.Card
+import android.content.Context
+import app.magic.wilson.zach.com.magicappkotlin.R
 import com.github.kittinunf.fuel.Fuel
-import com.github.kittinunf.fuel.core.FuelError
 import com.github.kittinunf.fuel.core.Request
-import com.github.kittinunf.fuel.core.Response
-import com.github.kittinunf.result.Result
-import java.sql.ResultSet
+import java.util.Locale
 
 /**
- * Created by Zach on 2/10/18.
+ * The API Client to get card data from the server.
  */
 
+fun getCards(context: Context, color: String?, rarity: String?) : Request {
+    // deviceLanguage is the language code
+    val deviceLanguage = Locale.getDefault().language
 
-fun getCards(color: String?, rarity: String?) : Request {
-    return Fuel.get("/mtg/cards", listOf("language" to "en", "colors" to "red"))
+    // validLanguages is a predetermined list of languages the app has data for and can support
+    val validLanguages = context.resources.getStringArray(R.array.card_languages_array)
+
+    // if the language is supported, then filter the card results to that language.
+    // Otherwise default to English
+    if (validLanguages.contains(deviceLanguage)) {
```

Comment:
I prefer to do this instead of having 2 return statements. It's best practice for you functions to only ever have 1 return statement. Replace these lines with this 


```
val filterLanguage = if (!validLanguages.contains(deviceLanguage)) deviceLanguage else "en"
return Fuel.get("/mtg/cards", listOf("language" to "en", "colors" to color, "rarity" to rarity))
```